### PR TITLE
ocamlPackages.lustre-v6: 6.103.3 -> 6.107.1

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/lustre-v6/default.nix [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
+{ lib, buildDunePackage, fetchurl, extlib, lutils, rdbg, yaml }:
 
 buildDunePackage rec {
   pname = "lustre-v6";
-  version = "6.103.3";
+  version = "6.107.1";
 
   useDune2 = true;
 
   minimalOCamlVersion = "4.05";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/lustre-v6/default.nix:10]:
```suggestion
  minimalOCamlVersion = "4.12";

```

Comment (wegank) [reply pkgs/development/ocaml-modules/lustre-v6/default.nix]: Done.

**Comment by vbgl** [comment]
It seems that the hash of the source tarball is wrong (something may have changed outside of nixpkgs).

**Comment by wegank** [comment]
Oh my, the tarball is updated [this morning](https://www-verimag.imag.fr/DIST-TOOLS/SYNCHRONE/pool/)...

**Pull request merged**